### PR TITLE
fix: Upgrade sqlalchemy from 1.x to 2.x regarding PVE-2022-51668.

### LINKED ISSUE #4062: SQLAlchemy 2 compatibility
**Is your feature request related to a problem? Please describe.**

There is a security vulnerability PVE-2022-51668 in SQLAlchemy < 2.0.0.b1.
https://data.safetycli.com/v/51668/f17/

However feast pinned the SQLAlchemy version to be < 2. We are struggling to have solve this vulnerability in our feast dependent projects. 
https://github.com/feast-dev/feast/blob/master/setup.py#L60

**Describe the solution you'd like**

Change the version specification to be >1 only

**Describe alternatives you've considered**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by shuchu**
**What this PR does / why we need it**:

Upgrade the SqlAlchemy to 2.x Regarding the PVE-2022-51668  mentioned in issue: [#4062 ](https://github.com/feast-dev/feast/issues/4062)

Implemented following the instruction: https://docs.sqlalchemy.org/en/20/changelog/migration_20.html

**Which issue(s) this PR fixes**:
Fixes #4062

### CODE REVIEW

File: sdk/python/feast/infra/registry/sql.py [hunk 1]
```diff
@@ -165,6 +165,8 @@ class FeastMetadataKeys(Enum):
 
 logger = logging.getLogger(__name__)
 
+CachingRegistry
```
Comment (sudohainguyen) [inline sdk/python/feast/infra/registry/sql.py:168]: is this typo? 🤔

Comment (franciscojavierarceo) [inline sdk/python/feast/infra/registry/sql.py:168]:
```suggestion
CacheRegistry
```

Comment (franciscojavierarceo) [reply sdk/python/feast/infra/registry/sql.py:168]: +1

**Comment by jeremyary** [comment]
@sudohainguyen @franciscojavierarceo @tokoko  any further concerns here? Would be good to fold this vuln fix into release

**Review by franciscojavierarceo — APPROVED** [review]
lgtm

**Pull request merged**